driver/normalizer/strconv.go

```diff
@@ -75,12 +75,15 @@ func replaceEscapedMaybe(s, old, repl string) string {
 		out.WriteString(s[:pos])
 		s = s[pos+len(old):]
 		r, n := utf8.DecodeRuneInString(s)
-		s = s[n:]
 		if r >= '0' && r <= '9' {
 			out.WriteString(old)
 		} else {
 			out.WriteString(repl)
 		}
+		if strings.IndexRune(old, r) == 0 { // old startsWith r
```

Comment:
`IndexRune` is a bit slow. As an alternative, you can extract the first rune outside of the loop and compare it with `r` here.

Reply:
Thanks for sharing an assumption and speedup suggestion!
Could you please also share the expected speed as well?

Reply:
Agreed: And in the common case where `old` does _not_ begin with `r`, this scans the entire string. 

As @dennwc suggests, this can be made:

```go
oldFirst, _ := utf8.DecodeRuneInString(old)
…
if oldFirst == r {
  continue
}
```

This ensures you never scan _further_ than the first rune, even if it isn't the needle.

Reply:
But that would also make some cases slower as we now need to always decode a rune.
Would be interested to know the expected performance implications..

Reply:
In cases where the string has nontrivial length, the cost impact is obviously smaller to decode one rune outside the loop. For strings that are very short we might lose a few cycles, but in that case we don't really care, because the string isn't that long.

I don't think this requires a benchmark—it's a straightforward consequence of not scanning the whole input when we don't need to.

Reply:
Checked to quantify what means small from initial suggestion - seems like ~6ns/op but indeed, O(n) as the worst case does not sound good!

Before
```
BenchmarkReplacingNullEscape/#00-4         	30000000	        47.1 ns/op	       8 B/op	       1 allocs/op
BenchmarkReplacingNullEscape/#01-4         	30000000	        47.6 ns/op	       8 B/op	       1 allocs/op
BenchmarkReplacingNullEscape/#02-4         	20000000	        79.8 ns/op	       8 B/op	       1 allocs/op
BenchmarkReplacingNullEscape/#03-4         	10000000	       174 ns/op	      24 B/op	       2 allocs/op
BenchmarkReplacingNullEscape/#04-4         	10000000	       218 ns/op	      56 B/op	       3 allocs/op
BenchmarkReplacingNullEscape/#05-4         	20000000	       103 ns/op	      16 B/op	       1 allocs/op
BenchmarkReplacingNullEscape/#06-4         	10000000	       135 ns/op	       8 B/op	       1 allocs/op
```

After the suggested change
```
BenchmarkReplacingNullEscape/#00-4         	30000000	        49.1 ns/op	       8 B/op	       1 allocs/op
BenchmarkReplacingNullEscape/#01-4         	30000000	        52.4 ns/op	       8 B/op	       1 allocs/op
BenchmarkReplacingNullEscape/#02-4         	20000000	        78.9 ns/op	       8 B/op	       1 allocs/op
BenchmarkReplacingNullEscape/#03-4         	10000000	       160 ns/op	      24 B/op	       2 allocs/op
BenchmarkReplacingNullEscape/#04-4         	10000000	       208 ns/op	      56 B/op	       3 allocs/op
BenchmarkReplacingNullEscape/#05-4         	20000000	        93.5 ns/op	      16 B/op	       1 allocs/op
BenchmarkReplacingNullEscape/#06-4         	20000000	       113 ns/op	       8 B/op	       1 allocs/op
```

Reply:
Addressed in fd6d585

Reply:
FYI, the [`benchcmp`](https://godoc.org/golang.org/x/tools/cmd/benchcmp) tool is quite useful for comparing benchmark results.